realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -443,18 +433,20 @@ protected final void copy(ColumnInfo rawSrc, ColumnInfo rawDst) {
 
         proxyState.getRealm$realm().checkIfValid();
         OsList osList = proxyState.getRow$realm().getModelList(columnInfo.columnRealmListIndex);
-        osList.removeAll();
-        if (value == null) {
-            return;
-        }
-        for (RealmModel linkedObject : value) {
-            if (!(RealmObject.isManaged(linkedObject) && RealmObject.isValid(linkedObject))) {
-                throw new IllegalArgumentException("Each element of 'value' must be a valid managed object.");
+        if (value != null && value.size() == osList.size()) {
+            for (int i = 0; i < value.size(); i++) {
```

Comment:
it would be faster to use a temp var to store the size to avoid multiple JNI calls here.